core/jvm/src/main/scala/fs2/interop/flow/package.scala

```diff
@@ -0,0 +1,257 @@
+    *                   A high number can be useful if the publisher is triggering from IO,
+    *                   like requesting elements from a database.
+    *                   The publisher can use this `bufferSize` to query elements in batch.
+    *                   A high number will also lead to more elements in memory.
+    * @param onSubscribe The effectual function to run during the subscription process,
+    *                    receives a [[Subscriber]] that should be used to subscribe to a [[Publisher]].
+    *                    The `subscribe` operation must be called exactly once.
+    */
+  def fromSubscriber[F[_], A](
+      bufferSize: Int
+  )(
+      subscribe: Subscriber[A] => F[Unit]
+  )(implicit
+      F: Async[F]
+  ): Stream[F, A] =
+    Stream
+      .eval(StreamSubscriber[F, A](bufferSize))
+      .flatMap { subscriber =>
+        subscriber.stream(
+          F.uncancelable(_ => subscribe(subscriber))
+        )
+      }
+
+  /** Creates a [[Stream]] from an `onSubscribe` effectual function.
+    *
+    * This function is useful when you need to perform some effectual actions during the subscription process.
+    * The subscription process is uncancelable.
+    *
+    * @example {{{
+    * scala> import cats.effect.IO
+    * scala> import fs2.Stream
+    * scala> import fs2.interop.flow.fromOnSubscribe
+    * scala> import java.util.concurrent.Flow.Publisher
+    * scala>
+    * scala> // Interop with the third party library.
+    * scala> def thirdPartyPublisher(): Publisher[Int] = ???
+    * scala> fromOnSubscribe[IO, Int](bufferSize = 16) { subscribe =>
+    *      |   IO.println("Subscribing!") >>
+    *      |   subscribe(thirdPartyPublisher()) >>
+    *      |   IO.println("Subscribed!")
+    *      | }
+    * res0: Stream[IO, Int] = Stream(..)
+    * }}}
+    *
+    * @note The publisher only receives a subscriber when the stream is run.
+    *
+    * @see [[fromPublisher]] for a simpler version that only requires a [[Publisher]].
+    *
+    * @param bufferSize setup the number of elements asked each time from the [[Publisher]].
+    *                   A high number can be useful if the publisher is triggering from IO,
+    *                   like requesting elements from a database.
+    *                   The publisher can use this `bufferSize` to query elements in batch.
+    *                   A high number will also lead to more elements in memory.
+    * @param onSubscribe The effectual action to run during the subscription process,
+    *                    represented as a function that receives as an argument
+    *                    the actual `subscribe` operation; as another function receiving the [[Publisher]].
+    *                    The `subscribe` operation must be called exactly once.
+    */
+  def fromOnSubscribe[F[_], A](
+      bufferSize: Int
+  )(
+      onSubscribe: (Publisher[A] => F[Unit]) => F[Unit]
+  )(implicit
+      F: Async[F]
+  ): Stream[F, A] =
```

Comment:
Sorry, still 👎 on this method. As I said, the signature is very confusing (as callbacks are) and it's not isomorphic to any existing interface in the Reactive Streams API.

Reply:
I ended up removing it because it was redundant and still unsafe.